utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -224,7 +223,7 @@ def host_install_destdir(self, host_target):
                 # If this is one of the hosts we should lipo,
                 # install in to a temporary subdirectory.
                 return '%s/intermediate-install/%s' % \
-                    (self.args.install_destdir, host_target)
+                    (os.path.dirname(self.build_dir), host_target)
```

Comment:
Found it, the issue was that I wasn't taking the `dirname` of the build directory, like the method I changed above did to get the `build_root`.

Ready for another mac toolchain CI run, @benlangmuir.

Reply:
I so very much wish I had seen this before tracking down the same bug